connection.go

```diff
@@ -605,6 +622,112 @@ func (sc *snowflakeConn) getQueryResult(ctx context.Context, resultPath string)
 	return respd, nil
 }
 
+// checkQueryStatus return error - nil means the query completed successfully - there is query result to fetch.
+// when check the status too soon there may not be status available, an ErrQueryStatus will be returned.
```

Comment:
A query goes through multiple stages:
- first, it can be queued for execution waiting on warehouse
- when it is executing, it may not have result yet
- then query can complete with result or error

Is `ErrQueryStatus` returned when the query is currently queued for execution?
And is `ErrQueryIsRunning 279301` when it is executing and have no result yet?

I think we can be more specific with the comment `when check the status too soon...`